### PR TITLE
Add 'ignore match' functionality to UI

### PR DESCRIPTION
**Pull request by jonathonherbert**
NB: please review and merge #87 first! it splits the state legwork into a separate PR to make this more readable.

## What does this change?

Building on #87, this PR adds the 'ignore match' command to the UI.

tooltip-ignore

I've used material-ui to assist with buttons and tooltips here – the icons are already used in our other tooling, and I didn't want to reinvent the tooltip wheel when an implementation was already available in the package!

However, there are some problems with the preact implementation that prevent us using the icon components directly for now. I've used a copy-paste here for expediency, with a comment to let others know that this isn't desirable in the long term.

## How to test

In the plugin UI, clicking 'ignore this match' should remove the match from the document and the results pane.

## How can we measure success?

As above.

### CODE REVIEW

File: build/rollup.config.dev.js [hunk 1]
```diff
@@ -15,6 +18,15 @@ export default [
     },
     plugins: [
       ...defaultPlugins,
+      alias({
+        entries: {
+          react: "preact/compat",
```
Comment (jonathonherbert) [inline build/rollup.config.dev.js:23]: We must alias `preact` to ensure material-ui imports `preact`, not `react`, when calling React-related functions. `preact` has a compatibility layer with an identical interface.

File: package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "pages": "npm run build && http-server ./pages/dist",
     "prepublishOnly": "npm run build && tsc -p tsconfig.declaration.json",
     "build": "rollup -c build/rollup.config.build.js",
-    "watch": "rollup -w -c build/rollup.config.dev.js",
+    "watch": "rollup -w -m -c build/rollup.config.dev.js",
```
Comment (jonathonherbert) [inline package.json:21]: Add sourcemaps to aid debugging.

File: src/ts/components/Match.tsx [hunk 3]
```diff
@@ -36,6 +41,20 @@ class Match<TMatch extends IMatch> extends Component<IMatchProps<TMatch>> {
       matchContext,
       markAsCorrect
     };
+
+    const suggestionsToRender = replacement ? [replacement] : suggestions || [];
```
Comment (jonathonherbert) [inline src/ts/components/Match.tsx:45]: A simple move to colocate var declarations.

File: src/ts/components/Match.tsx [hunk 4]
```diff
@@ -48,19 +67,33 @@ class Match<TMatch extends IMatch> extends Component<IMatchProps<TMatch>> {
...
+                  href={this.getFeedbackLink(
+                    this.props.feedbackHref!,
+                    feedbackInfo
+                  )}
+                >
+                  Issue with this result? Tell us!
+                </a>
+              </div>
+            )}
+            <div className="MatchWidget__ignore-match">
```
Comment (jonathonherbert) [inline src/ts/components/Match.tsx:84]: The _actual_ tooltip and button markup!

Comment (aug24) [reply src/ts/components/Match.tsx:84]: Surely this shouldn't be displayed unless onIgnoreMatch is defined?

Comment (jonathonherbert) [reply src/ts/components/Match.tsx]: You're right! I've just made that change.

**Pull request merged**